container/internal/aws/metrics.go

```diff
@@ -0,0 +1,118 @@
+// Package aws provides AWS service integrations for metrics and monitoring.
+package aws
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+const (
+	// MetricsNamespace is the CloudWatch namespace for log forwarding metrics
+	MetricsNamespace = "HCPLF/LogForwarding"
+)
+
+// MetricsPublisher handles publishing metrics to CloudWatch
+type MetricsPublisher struct {
+	client *cloudwatch.Client
+	logger *slog.Logger
+}
+
+// NewMetricsPublisher creates a new metrics publisher
+func NewMetricsPublisher(client *cloudwatch.Client, logger *slog.Logger) *MetricsPublisher {
+	return &MetricsPublisher{
+		client: client,
+		logger: logger,
+	}
+}
+
+// PushMetrics publishes metrics to CloudWatch
+// method is either "cloudwatch" or "s3"
+// metricsData is a map of metric names to values (e.g., {"successful_delivery": 1, "failed_delivery": 0})
+func (p *MetricsPublisher) PushMetrics(ctx context.Context, tenantID, method string, metricsData map[string]float64) error {
+	if len(metricsData) == 0 {
+		p.logger.Debug("no metrics to push")
+		return nil
+	}
+
+	metricData := make([]types.MetricDatum, 0, len(metricsData))
```

Comment:
minor nit: `metricData` and `metricsData` are named pretty similarly - would it make sense to differentiate them a bit more? ie - 
```suggestion
	metrics := make([]types.MetricDatum, 0, len(metricsData))
```

Reply:
Agree!